libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -1466,6 +1466,21 @@ bool AP_Arming::serial_protocol_checks(bool display_failure)
     return true;
 }
 
+//Check for estop
+bool AP_Arming::estop_checks(bool display_failure)
+{
+     const RC_Channel *chan = rc().find_channel_for_option(RC_Channel::AUX_FUNC::ARM_EMERGENCY_STOP);
+     bool arm_estop_sw_low;
+     if (chan != nullptr && chan->get_aux_switch_pos() == RC_Channel::AuxSwitchPos::LOW) {
+        arm_estop_sw_low = true;
+     }
+     if (SRV_Channels::get_emergency_stop() && !arm_estop_sw_low) {
+        check_failed(display_failure,"Motors Emergency Stopped");
+        return false;
+     }
+     return true;
```

Comment:
```suggestion
    if (!SRV_Channels::get_emergency_stop()) {
        // not emergency-stopped, so no prearm failure:
        return true;
    }
    // vehicle is emergency-stopped; if this *appears* to have been done via switch then we do not fail prearms:
    const RC_Channel *chan = rc().find_channel_for_option(RC_Channel::AUX_FUNC::ARM_EMERGENCY_STOP);
    if (chan == nullptr) {
        // RC channel not configured for option, so estop not activated via switch
        return true;  // no prearm failure
    }
    if (chan->get_aux_switch_pos() != RC_Channel::AuxSwitchPos::HIGH) {
         // switch is not in "asserted" position, so likely not reason we are estopped
         return true;  // no prearm failure
    }   
    check_failed(display_failure,"Motors Emergency Stopped");
    return false;
```

 - arm_estop_sw_low was uninitialised stack data - you need to assign a value to a stack variable before testing it.
 - we can avoid some work when we're not estopped - 99.99% of the time this will be the case.

If we do get reports of this not working we could potentially keep around an "estop reason"....

Reply:
updated

Reply:
made changes to get the desired effect

Reply:
I had already fixed that